File: INFS4102_project-main/Frontend/eshop_ui/lib/models/AccessoryProduct.dart
import 'Product.dart';

class AccessoryProduct extends Product {
  final String material;

  const AccessoryProduct({
    required super.id,
    required super.name,
    required super.price,
    required super.category,
    required super.desc,
    required super.imageURL,
    required this.material,
  });

  //converts json
  factory AccessoryProduct.fromJson(Map<String, dynamic> json) {
    return AccessoryProduct(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      price: (json['price'] as num).toDouble(),
      category: json['category'] ?? '',
      desc: json['desc'] ?? '',
      imageURL: json['imageURL'] ?? '',
      material: json['material'] ?? '',
    );
  }
}

File: INFS4102_project-main/Backend/eshop/src/main/java/qa/udst/eshop/models/AccessoryProduct.java
package qa.udst.eshop.models;

import jakarta.validation.constraints.*;

//Accessory Model: + Material
public class AccessoryProduct extends Product{
    @NotBlank
    private String material;

    public AccessoryProduct(String name, double price, ProductCategory category, String desc,
            @NotBlank String material, @NotBlank String imageURL) {
        super(name, price, category, desc, imageURL);
        this.material = material;
    }

    public void setMaterial(String material) {
        this.material = material;
    }

    public String getMaterial() {
        return material;
    }


}
